src/redux/books/books.js

```diff
@@ -0,0 +1,29 @@
+// Define action types for adding and removing a book.
+const ADD_BOOK = 'bookstore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+
+const initialState = [];
+
+export const addBook = (book) => ({
+  type: ADD_BOOK,
+  payload: book,
+});
+
+export const removeBook = (index) => ({
+  type: REMOVE_BOOK,
+  payload: index,
+});
+
+const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return { state: [...state.state, action.payload] };
+    case REMOVE_BOOK:
+      return [...state.slice(0, action.index),
+        ...state.slice(action.index + 1), state.length];
```

Comment:
- [ ] [  _Optional_ ] For this method, using the filter method would be a much better choice with way less time complexity. The logic would also be much simpler as you can see below.
``` js
state.filter((book) => book.id !== action.index);
```

Reply:
OH tHANKS